### PR TITLE
💾 Update Data Model 

### PR DESCRIPTION
**Pull request by jbethanco**
## 💬 Discussion 

Based on the updated requirements for upload to scheduling; made updates to the data model and encodable to support new information flow. Also adds training event names to exported training events as requested by the PBS team. Does not store them with crewLine in MissionEventRecords. It is used only for export until PBS has the tables implemented on their end.
(See: [PBS-1589](https://jira.il2.dso.mil/browse/PBS-1589))

## 📕 Type
- [ ] Fix (change which fixes an issue)
- [X] Feature (change which adds functionality)
- [ ] Refactor (updates existing code for performance, readability, style, architecture) 
- [ ] Chore (a change that does not change production code) 

## 🧾 Checklist
- [X] Has a corresponding task in Jira
- [X] Unit and UI tests pass locally with my changes
- [X] Unit and UI tests pass remotely with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works

#### Adds  ✏️ 
* new properties to Mission Entity:
isReadyForReview
organizationID
pbsID
* convenience UUID setter functions for UUIDs associated with Mission
* pbsID, organizationID, and isReadyForReview to Mission Encodable extension
* logic to fetch name for eventID to MissionEventRecord+Encodable
* name encoding field to MissionEventRecord+Encodable
* set pbsID with downloaded event.id
* organization ID to missions upon import
* sortedKeys output formatting to JSONEncoder so exported JSON looks slightly better

#### Removes  🔪  
* overwrite of Mission id with PBS ID on download

### CODE REVIEW

**Comment by bmorris-revacomm** [comment]
whooooaaaaa didn't know you could do that lol

File: PBLogging/Data/DataController.swift [hunk 1]
```diff
@@ -89,7 +89,7 @@ class DataController: ObservableObject {
         
         let mission = Mission(context: container.viewContext)
         mission.name = pbsEvent.name
-        mission.id = UUID(uuidString: pbsEvent.id)
+        mission.setPBSid(with: pbsEvent.id)
```
Comment (bmorris-revacomm) [inline PBLogging/Data/DataController.swift:92]: Curious as to why you changed it to a function? 🤔

File: PBLogging/Data/EntityEncodableExtensions/MissionEventRecord+Encodable.swift [hunk 2]
```diff
@@ -5,18 +5,29 @@
...
 
+import CoreData
 import Foundation
 
 extension MissionEventRecord: Encodable {
     
     private enum CodingKeys: String, CodingKey {
-        case asInstructor, eventID, numberAccomplished
+        case asInstructor, eventID, numberAccomplished, name
     }
```
Comment (bmorris-revacomm) [inline PBLogging/Data/EntityEncodableExtensions/MissionEventRecord+Encodable.swift:15]: yayyyyyyyy

**Review by bmorris-revacomm — COMMENTED** [review]
I don't have a way to look at the imports view, so I can't really check it as well as I would like, but just looking at the code, it seems to be good 👍

File: PBLogging/Data/EntityExtensions/Mission+Accessors.swift [hunk 3]
```diff
@@ -41,4 +41,11 @@ extension Mission {
         if takeoffDates.isEmpty { return nil }
         return takeoffDates.min()
     }
+    
+    func setPBSid(with string: String) {
+        pbsID = UUID(uuidString: string)
+    }
+    func setOrganizationID(with string: String) {
+        organizationID = UUID(uuidString: string)
+    }
```
Comment (mcgarrys) [inline PBLogging/Data/EntityExtensions/Mission+Accessors.swift:50]: Could update the Mission+Accessors tests with these new setters.

**Review by mcgarrys — COMMENTED** [review]
Challenge Accepted 😈 , this is one beautiful PR. Didn't know you could do the check boxes!

Like Beth, I also can't test the import. But code looks clean. Left a comment regarding tests, because I have them on my mind now lol.

File: PBLogging/Data/DataController.swift (hunk 1 above)
Comment (jbethanco) [reply PBLogging/Data/DataController.swift:92]: I thought I was being a good citizen and abstracting away some complexity of going from the received string to a UUID, but now that I'm looking at it side by side, it doesn't seem like much of a benefit. I think I'll revert that one.

File: PBLogging/Data/EntityExtensions/Mission+Accessors.swift (hunk 3 above)
Comment (jbethanco) [reply PBLogging/Data/EntityExtensions/Mission+Accessors.swift:50]: Will do.

**Review by petermisik — APPROVED** [review]
Looks awesome and works!! 🔥 🤘 🔥

File: PBLogging/Data/EntityExtensions/Mission+Accessors.swift (hunk 3 above)
Comment (jbethanco) [reply PBLogging/Data/EntityExtensions/Mission+Accessors.swift]: Wound up removing the accessors.

**Pull request merged**